test/test_tensor_creation_ops.py

```diff
@@ -712,8 +711,9 @@ def test_cat_different_devices(self, devices):
     def test_cat_stack_cross_devices(self, device):
         cuda = torch.randn((3, 3), device=device)
         cpu = torch.randn((3, 3), device='cpu')
-        out_cpu = cpu.clone()
         out_cuda = cuda.clone()
```

Comment:
I think this variable and the related test can be removed, too (out= will test for cross-device input x out=)

Reply:
There might be a new out= test that needs to be added though to account for cat's memory layout behavior. Is there a test that:

- if out= is the correct shape then the memory format of out= is respected
- if out= is not the correct shape then the memory format is that of the first tensor? 

That's not tested by Opinfo-based out= testing and so needs to be covered specially

Reply:
> I think this variable and the related test can be removed, too (out= will test for cross-device input x out=)
Done!